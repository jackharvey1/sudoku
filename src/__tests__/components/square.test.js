import React from 'react';
import Square from '../../components/square';
import ShallowRenderer from 'react-test-renderer/shallow';

let renderer;

beforeEach(() => {
    renderer = new ShallowRenderer();
});

describe('Square component', () => {
    it('renders as expected in basic state', () => {
        renderer.render(<Square
            value={1}
            onClick={() => {}}
            box={0}
            square={0}
            isSelected={false}
            isLocked={false}
        />);
        const output = renderer.getRenderOutput();
        expect(output).toMatchSnapshot();
    });

    it('renders with bold text when locked', () => {
        renderer.render(<Square
            value={1}
            onClick={() => {}}
            box={0}
            square={0}
            isSelected={false}
            isLocked={true}
        />);
        const output = renderer.getRenderOutput();
        expect(output).toMatchSnapshot();
    });

    it('renders with background when selected', () => {
        renderer.render(<Square
            value={1}
            onClick={() => {}}
            box={0}
            square={0}
            isSelected={true}
            isLocked={false}
        />);
        const output = renderer.getRenderOutput();
        expect(output).toMatchSnapshot();
    });
});
