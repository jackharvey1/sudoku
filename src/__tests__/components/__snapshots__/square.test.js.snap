// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square component renders as expected in basic state 1`] = `
<div
  box={0}
  className="css-iz9zke"
  onClick={[Function]}
  square={0}
  value={1}
>
  <p
    className="css-1h92lle"
  >
    1
  </p>
</div>
`;

exports[`Square component renders with background when selected 1`] = `
<div
  box={0}
  className="css-9v78r0"
  onClick={[Function]}
  square={0}
  value={1}
>
  <p
    className="css-1h92lle"
  >
    1
  </p>
</div>
`;

exports[`Square component renders with bold text when locked 1`] = `
<div
  box={0}
  className="css-14zxudu"
  onClick={[Function]}
  square={0}
  value={1}
>
  <p
    className="css-1h92lle"
  >
    1
  </p>
</div>
`;
